src/python/pants/option/parser.py

```diff
@@ -538,18 +538,17 @@ def _existing_scope(self, arg):
   _ENV_SANITIZER_RE = re.compile(r'[.-]')
 
   @staticmethod
-  def parse_dest(*args, **kwargs):
-    """Select the dest name for an option registration.
+  def parse_name_and_dest(*args, **kwargs):
+    """Return the name and dest for an option registration.
 
     If an explicit `dest` is specified, returns that and otherwise derives a default from the
     option flags where '--foo-bar' -> 'foo_bar' and '-x' -> 'x'.
     """
-    explicit_dest = kwargs.get('dest')
-    if explicit_dest:
-      return explicit_dest
-
     arg = next((a for a in args if a.startswith('--')), args[0])
-    return arg.lstrip('-').replace('-', '_')
+    name = arg.lstrip('-').replace('-', '_')
+
+    dest = kwargs.get('dest')
+    return (name, dest if dest else name)
```

Comment:
Nit:

```python
dest = kwargs.get('dest', name)
return name, dest
```

Also note that the parentheses are not necessary when returning a tuple.

Reply:
That isn't quite equivalent, because `dest=None` would still be used.